source/codegen/templates/proto_helpers.mako

```diff
@@ -30,16 +30,22 @@ enum ${service_class_prefix}Attributes {
 <%
   enum = enums[enum_name]
   enum_value_prefix = enum.get("enum-value-prefix", common_helpers.pascal_to_snake(enum_name).upper())
+  is_mapped_enum = enum.get("generate-mappings", False)
+  enum_name = f"{enum_name}Mapped" if (is_mapped_enum and not enum_name.endswith("Mapped")) else enum_name
```

Comment:
Why is this conditional needed? Can you comment it? And maybe the corresponding logical case in metadata mutations? The `generate-mappings` vs `{enum_name}Mapped` is already a little tricky to follow.

Reply:
I don't want the code generator to modify the names of enums. If the name of the enum needs to have "mapped" in it, we should do that in the `enums.py`.

Reply:
Problem is, if we are asking developers to add "Mapped" suffix in `enums.py` some of our consolidated enums might have "Mapped" keyword twice in the constant name. 

If equivalent of below from enums.py is enum associated with attribute...
```
enum OsNames {
  VAL_ANDROID: "ANDROID"
  VAL_IOS: "IOS"
}
```

...consolidated string enum for above in `.proto` file would currently look like
```
enum NiDmmStringAttributeValuesMapped {
    NIDMM_MAPPED_UNSPECIFIED: 0,
    NIDMM_OS_NAMES_ANDROID: 1,
    NIDMM_OS_NAMES_IOS: 2
}
```
But if developers add `Mapped` to `OsNames` enum, it would look like...
```
enum NiDmmStringAttributeValuesMapped {
    NIDMM_MAPPED_UNSPECIFIED: 0,
    NIDMM_OS_NAMES_MAPPED_ANDROID: 1,
    NIDMM_OS_NAMES_MAPPED_IOS: 2
}
```
or worse, based on discussion about how to add `_MAPPED_` for `_UNSPECIFIED` value [here](https://github.com/ni/grpc-device/pull/224/files/5f3df308d10ecd15d240eafe7cd01ef3cc9e8fee#r655863859)
```
enum NiDmmStringAttributeValuesMapped {
    NIDMM_MAPPED_UNSPECIFIED: 0,
    NIDMM_MAPPED_OS_NAMES_MAPPED_ANDROID: 1,
    NIDMM_MAPPED_OS_NAMES_MAPPED_IOS: 2
}
```
Adding `Mapped` suffix to all enums with `generate-mapping` tag during mutation/codegen has advantage that developers just have to tag enum with above attribute to say that mappings should be generated and gives logical name (that comes from base metadata for MI enums) for the enum and our code generator will consistently add same suffix to all such enums (even for mapped enums in DAQmx and so on).

Reply:
I still don't see much benefit in putting "Mapped" at the end of the type name unless we're talking about the attribute value enums.

If a normal function refers to a mapped enum, we should see this:
```
oneof {
  double foo_raw = 2,
  MyEnum foo_enum_mapped = 3
}
```
When we merge enums for attributes, we'll end up with this:
_(note: I'm not trying to be precise with names here, I just don't have the real naming scheme in front of me while writing the example)_
```
enum NiDmmReal64AttributeValues {
  ...
}
enum NiDmmReal64AttributeMappedValues {
  ...
}
message NiDmmSetViReal64AttributeRequest {
  ...
  oneof {
    double value_raw = 5,
    NiDmmReal64AttributeValues value_enum = 6,
    NiDmmReal64AttributeMappedValues value_enum_mapped = 7
  }
}
```